vmdb/app/models/ems_kubernetes.rb

```diff
@@ -11,19 +11,40 @@ def self.description
     @description ||= "Kubernetes".freeze
   end
 
-  def self.raw_connect(hostname, port, api_version)
+  def self.raw_connect(uri, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    Kubeclient::Client.new(uri, api_version)
+  end
 
+  def api_endpoint
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
+    uri = URI::HTTP.build(:path => '/api', :port => port.to_i)
     uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    uri.to_s
```

Comment:
I don't mind this being split this out into its own method, but it should just return the URI object, not the string.  kubeclient should be changed to support a URI object or a String (already discussed with @abonas in the past).